### PR TITLE
ddl: remove unused code

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve?

Ref：https://github.com/pingcap/tidb/issues/23013 https://github.com/pingcap/tidb/issues/22979  

Problem Summary:

### What is changed and how it works?

make the lint `unused`  pass in the ddl package.

What's Changed:

How it Works:

Tests 

- Unit test

### Release note 

- No

### CODE REVIEW

File: ddl/error.go [hunk 1]
```diff
@@ -53,15 +53,16 @@ var (
...
+	// nolint:unused
+	errUnknownTypeLength        = dbterror.ClassDDL.NewStd(mysql.ErrUnknownTypeLength)
+	errInvalidDDLJobVersion     = dbterror.ClassDDL.NewStd(mysql.ErrInvalidDDLJobVersion)
+	errInvalidUseOfNull         = dbterror.ClassDDL.NewStd(mysql.ErrInvalidUseOfNull)
+	errTooManyFields            = dbterror.ClassDDL.NewStd(mysql.ErrTooManyFields)
+	errTooManyKeys              = dbterror.ClassDDL.NewStd(mysql.ErrTooManyKeys)
+	errInvalidSplitRegionRanges = dbterror.ClassDDL.NewStd(mysql.ErrInvalidSplitRegionRanges)
+	errReorgPanic               = dbterror.ClassDDL.NewStd(mysql.ErrReorgPanic)
+	errFkColumnCannotDrop       = dbterror.ClassDDL.NewStd(mysql.ErrFkColumnCannotDrop)
+	errFKIncompatibleColumns    = dbterror.ClassDDL.NewStd(mysql.ErrFKIncompatibleColumns)
```
Comment (zimulala) [inline ddl/error.go:65]: Could we remove these variables?

Comment (nolouch) [reply ddl/error.go:65]: I'm not sure, the error convert from mysql. should we remove it?

Comment (tangenta) [reply ddl/error.go:65]: +1 We can add them on demand.

**Review by xiongjiwei — APPROVED** [review]

**Pull request merged**